### DISCUSSION TITLE
Amazon backport provider conflicts with boto3/botocore

### DISCUSSION THREAD

**Discussion by ring-pete** [apache/airflow#19236] [Q&A] [answered]
Hello,
I am standing up a 1.10.15 cluster and noticed these errors while attempting to install the amazon backport provider:

```
apache-airflow-backport-providers-amazon 2021.3.3 requires boto3<1.16.0,>=1.15.0, but you'll have boto3 1.10.50 which is incompatible.
```

When looking at the constraints file for python3.7 I can verify that it requires versions of boto that are out of date for the backport providers (https://raw.githubusercontent.com/apache/airflow/constraints-1.10.15/constraints-3.7.txt):
```
boto3==1.10.50
boto==2.49.0
```
Are any other people out there using the amazon provider on 1.10.15?  Is so how have you handled working around this conflict?

**Comment by potiuk** [comment, marked as answer]
Backport providers were released independently from airflow releases and you should install them separately. This is pretty normal that Airflow 1.10 will have different "provider" depenencies as constraints because in 1.10 providers were 'embedded' in Airflow package.

Constraints were specifically used because they are not "hard" dependencies and only used during the installation when specified. So the best course of action is to:

a) install airflow 1.10 with constraints
b) in a separate step install backport provider packages that you need - without constraints.

**Comment by ring-pete** [comment]
Thanks for the heads up on this!  My concern was that I may introduce incompatibilities by allowing the backport-providers to update dependencies.  In this case it looks pretty minor, just some minor version bumps to the boto libs.

Or are you suggesting that we have one virtual-env which contains our airflow target version and another venv which contains the backport-providers for example?

**Reply by potiuk** [reply]
Separate venvs won't work. And yes there is a porential risk of incompatibilities - for example if you try to run new Amazon operatora from backports and old from airflow 1.10 at the same time, the old ones might nor work. There is no guarantee that you can get. 

You have to take the risk. 


I think you are taking far Greater Risk however already by not migrating to Airflow 2. Airflow 1.10 reached end of life in June and it will not receive any more fixes. Not even critical security fixes.

 Latest backports were released in March and since then there were i think 5 releases of Amazon providers ( none of which made it to backports because we ended up releasing backports 6 months ago). 

So if you are afraid if one dependency upgrade will cause some problems, i think you are worried about completely wrong thing.

Upgrade to Airflow 2 ASAP.

**Reply by ring-pete** [reply]
Thanks for the confirmation that I was understanding things correctly.  Regarding Airflow 2, this is what we are in the middle of doing hence us upgrading the 1.10.15 which is the "bridge" release before 2.0.

**Reply by potiuk** [reply]
Cool. Keep my fingers crossed : 🤞
